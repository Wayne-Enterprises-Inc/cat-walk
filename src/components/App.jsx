import React from 'react';
import ReactDom from 'react-dom';
import axios from 'axios';
import requests from '../lib/axiosPrefilter.js';


import CompareModal from "./CompareModal";
import YourOutfit from './YourOutfit'
import OutFitCreater from './OutFitCreater'
import RatingBreakdown from './RatingsReviews/RatingBreakdown.jsx';

import RelatedItems from './RelatedItems.jsx';



import Overview from './ProductOverview/Overview.jsx';

// testing out styled components
import styled from 'styled-components';

class App extends React.Component {
  constructor() {
    super();
    this.state = {
      relatedProducts: [],
      starData: 0,

      productId: 19089,
      reviewData: null,
    };

    //bind functions here
    this.handleStarData = this.handleStarData.bind(this);
    this.relatedSelectHandle = this.relatedSelectHandle.bind(this)
    this.getReviewInfo = this.getReviewInfo.bind(this)
  }
  //functions/handler section

  //api pull test
  // componentDidMount() {
  //   axios.get(requests.pullCart)
  //     .then(products => {
  //       console.log(products)
  //     })
  //     .catch(error => {
  //       console.error('Error pulling products: ', error)
  //     })
  // }

  handleStarData(data) {
    this.setState({
      starData: data
    })
  }

  relatedSelectHandle(event) {
    if (this.state.productId !== '') {
      console.log('event: ', event.target.attributes.value.nodeValue)
      this.setState({
        productId: ''
      }, () => {
        this.setState({
          productId: event.target.attributes.value.nodeValue
        })
      }, () => {
        console.log('updated productId: ', this.state.productId)
      })
    } else {
      this.setState({
        productId: event.target.attributes.value.nodeValue
      }, () => {
        console.log('updated productId: ', this.state.productId)
      })
    }
  }

  getReviewInfo(id) {
    axios.get(requests.pullReviews + `/meta/?product_id=${id}`)
      .then(reviews => {
        //console.log('All reviews: ', reviews.data.ratings)
        this.setState({
          reviewData: reviews
        })
      })
      .catch(err => {
        console.error('Error getting review info: ', err)
      })
  }

  componentDidMount() {
    this.getReviewInfo(this.state.productId)
  }

  render() {

    return (
      <SyledApp>
        <LogoBar>
        <i class="fas fa-tshirt" style={{
            fontWeight: 'bold',
            float: 'left',
            marginLeft: '20px',
            marginTop: '16px',
            marginBottom: '16px',
            backgroundColor: '#f8b500'}}></i>
          <p style={{
            fontWeight: 'bold',
            float: 'left',
            backgroundColor: '#f8b500'
        }}>Clothing Inc.</p>
        <input
        style= {{
          float: 'right',
          marginRight: '40px',
          marginTop: '15px'
        }}
        type='text'
        placeholder='Product Search...'

        ></input>
        </LogoBar>

        <Overview
          starData={this.state.starData}
          productId={this.state.productId}
          reviewData={this.state.reviewData}
        />

        <div>
          <RelatedItems
          productId={this.state.productId}
          getId={this.relatedSelectHandle}
          allProducts={this.state.relatedProducts}
          starData={this.state.starData}
          reviewData={this.state.reviewData}
           />
          </div>

        <YourOutfit allProducts={this.state.relatedProducts}
        reviewData={this.state.reviewData} starData={this.state.starData} />
        {/* <OutFitCreater allProducts={this.state.relatedProducts}/> */}


        <RatingBreakdown
        onRatingChange={this.handleStarData}
        productId={this.state.productId}
        reviewData={this.state.reviewData}
        />

      </SyledApp>
    )
  }
}
// experimenting with the styled-components package

const SyledApp = styled.div`
  font-family: helvetica;
  background: #f7f7f7;
  color: #393e46;
`

const LogoBar = styled.div`

left: 50px;
height: 50px;
background: #5c636e;
`;

export default App;