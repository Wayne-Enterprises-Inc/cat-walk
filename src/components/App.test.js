//import { TestScheduler } from '@jest/core';
//import { hasUncaughtExceptionCaptureCallback } from 'node:process';
import React from 'react';
import ReactDom from 'react-dom';
import 'babel-polyfill';

import { getQueriesForElement } from '@testing-library/dom';
import { render, fireEvent, cleanup, wait } from '@testing-library/react';

import mockAxios from 'Axios';

import renderer from 'react-test-renderer'

import App from './App.jsx';

afterEach(cleanup);

test("renders the App", () => {
  //renders component to Dom
  //This will be done for almost every test.
  const root = document.createElement('div');
  const { getByText, getByLabelText, getById } = render(<App />)

  // const { getByText, getByLabelText } = getQueriesForElement(root);

  expect(getByText('Clothing Inc.')).not.toBeNull();
  //these two are equivalent test
  getByText('Clothing Inc.')
  getByText('Related Items')
  getByText('RATINGS & REVIEWS')


  //expect(getByText('$')).not.toBeNull();

  // this method is improved upon by the getQueriesForElement
  //expect(root.querySelector('p').textContent).toBe('Clothing Inc.')
})

it('matches snapshot', () => {
  const tree = renderer.create(<App />).toJSON();

  expect(tree).toMatchSnapshot();
})

test("should print product name", async () => {

mockAxios.get(() => {
  Promise.resolve({
    data: {
      campus: "hr-rfe",
      category: "Jackets",
      created_at: "2021-02-23T19:24:34.450Z",
      default_price: "140.00",
      description: "The So Fatigues will wake you up and fit you in. This high energy camo will have you blending in to even the wildest surroundings.",
      id: 19089,
      name: "Camo Onesie",
      slogan: "Blend in to your crowd"
    }
  })
})


  const { getByText, getByLabelText } = render(<App />)

  // const arrow = document.querySelector('#arrow')
  // console.log('arrow: ', arrow)

  await wait(()=> getByText('Camo Onesie'))

  //expect(getByText('Camo Onesie')).not.toBeNull();
  //getById('arrow');

})