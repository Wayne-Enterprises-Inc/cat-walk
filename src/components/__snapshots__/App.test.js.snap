// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-fvxABq egHPNc"
>
  <div
    className="sc-fbyett dtGzSM"
  >
    <i
      class="fas fa-tshirt"
      style={
        Object {
          "backgroundColor": "#f8b500",
          "float": "left",
          "fontWeight": "bold",
          "marginBottom": "16px",
          "marginLeft": "20px",
          "marginTop": "16px",
        }
      }
    />
    <p
      style={
        Object {
          "backgroundColor": "#f8b500",
          "float": "left",
          "fontWeight": "bold",
        }
      }
    >
      Clothing Inc.
    </p>
    <input
      placeholder="Product Search..."
      style={
        Object {
          "float": "right",
          "marginRight": "40px",
          "marginTop": "15px",
        }
      }
      type="text"
    />
  </div>
  <div
    className="sc-dtMiey dsqDgF"
  >
    <div
      className="sc-jUotMc gHMKsv"
    >
      <div>
        loading...
      </div>
      <div
        className="sc-cZMLWB bXmjnX"
      >
        <div
          className="sc-fWCJfs gPPPIe"
        >
          <div
            style={
              Object {
                "padding": "10px",
                "width": "300px",
              }
            }
          >
            <div
              className="sc-jIkYaL ljRQvZ"
            >
              <div
                className="sc-ZOsLE hkrPdR"
              >
                <div
                  className="sc-hmjpBu tKtDQ"
                />
              </div>
            </div>
            <div
              className="sc-jOxuqd bkJvdE"
              onClick={[Function]}
            >
              Read all reviews
            </div>
            <h5>
              :
            </h5>
            <h2 />
            <h3 />
            <h2>
              Style &gt; 
            </h2>
            <div
              className="sc-GamvQ hPWtgl"
            />
          </div>
        </div>
        <div>
          <form>
            <select
              className="sc-bTfYlY eeHBmC"
              onChange={[Function]}
            >
              <option>
                Select a Size
              </option>
            </select>
            <select
              className="sc-kHOZQx jLhOie"
              disabled={true}
              onClick={[Function]}
            >
              <option>
                Qty: 0
              </option>
            </select>
            <br />
            <button
              className="sc-eLwHGX hAeRTH"
              disabled={true}
            >
              Add to Cart
            </button>
            <br />
            <button
              className="sc-hOGjNT crOBLz"
            />
          </form>
        </div>
      </div>
    </div>
    <div
      className="sc-jQrCkL bANUka"
    >
      <div>
        <em>
          <b />
        </em>
      </div>
      <br />
      <div />
    </div>
  </div>
  <div>
    <div>
      <div>
        <h4>
          Related Items
        </h4>
        <div>
          <div
            style={
              Object {
                "marginLeft": "25%",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
  <div>
    <button
      onClick={[Function]}
    >
      Add Outfit
    </button>
  </div>
  <div
    className="sc-jJoQpE fvZtPz"
  >
    <div
      className="sc-hiCivh hHeKsX"
    >
      RATINGS & REVIEWS
    </div>
    <div
      className="sc-gWXaA-D lkfinj"
    >
      <div
        className="sc-cidCJl hVeMbZ"
      >
        <div
          className="sc-jcFkyM haAsQZ"
        >
          N/A
        </div>
        <div
          className="sc-caiKgP qJiku"
        >
          <div
            className="sc-iUKrWq cWmQoJ"
          >
            <div
              className="sc-iAKVOt ilBttS"
            />
          </div>
        </div>
      </div>
      <div
        className="sc-efQUeY kqyVhz"
      >
        <div
          className="sc-jObXwK kjOkvX"
        />
        <div
          className="sc-cTApHj jfGrnB"
        >
          5 stars
        </div>
        <div
          className="sc-dPiKHq cSIFfP"
        >
          <div
            className="sc-bBHHQT fZrTSx"
            width={null}
          />
        </div>
        <div
          className="sc-jObXwK kjOkvX"
        />
        <div
          className="sc-cTApHj jfGrnB"
        >
          4 stars
        </div>
        <div
          className="sc-dPiKHq cSIFfP"
        >
          <div
            className="sc-bBHHQT fZrTSx"
            width={null}
          />
        </div>
        <div
          className="sc-jObXwK kjOkvX"
        />
        <div
          className="sc-cTApHj jfGrnB"
        >
          3 stars
        </div>
        <div
          className="sc-dPiKHq cSIFfP"
        >
          <div
            className="sc-bBHHQT fZrTSx"
            width={null}
          />
        </div>
        <div
          className="sc-jObXwK kjOkvX"
        />
        <div
          className="sc-cTApHj jfGrnB"
        >
          2 stars
        </div>
        <div
          className="sc-dPiKHq cSIFfP"
        >
          <div
            className="sc-bBHHQT fZrTSx"
            width={null}
          />
        </div>
        <div
          className="sc-jObXwK kjOkvX"
        />
        <div
          className="sc-cTApHj jfGrnB"
        >
          1 stars
        </div>
        <div
          className="sc-dPiKHq cSIFfP"
        >
          <div
            className="sc-bBHHQT fZrTSx"
            width={null}
          />
        </div>
      </div>
      <div
        className="sc-cNKpQo cuLYpS"
      >
        0
         of reviews recommended this product
      </div>
    </div>
    <div
      className="sc-cCcYRi jmpBhT"
    >
      <div
        className="sc-fotPbf kUUlHI"
      >
        <div
          className="sc-fFehDp blBGsh"
        >
           reviews, sorted by  
          <div
            className="sc-jrQzUz gwhoOm"
          >
            <div
              className="sc-kDThTU kEPsS"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="sc-iqsfdx cuTrvZ"
              >
                Relevant
                 ▼
              </div>
            </div>
          </div>
        </div>
        <div
          className="sc-bkkfTU kvjTOP"
        />
        <div
          className="sc-bYoCmx blNjdO"
        >
          <div
            className="sc-kLwgWK hlJEcv"
          >
            <button
              onClick={[Function]}
            >
              MORE REVIEWS
            </button>
          </div>
          <div
            className="sc-ikJzcn dsoCKz"
          >
            <button>
               ADD A REVIEW    +
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
